### PR TITLE
Btree node fetch try

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

File: kvlds/btree_node.c [hunk 1]
```diff
@@ -188,8 +189,8 @@ btree_node_mknode(struct btree * T, int type, int height, size_t nkeys,
  * complete, with the node locked.
```
Comment (gperciva) [inline kvlds/btree_node.c]: I just noticed that the comment hasn't been updated.  I'm fixing that now.

File: kvlds/btree_node.c [hunk 2]
```diff
@@ -44,6 +44,8 @@ struct descend {
 
 static int callback_fetch(void *, int, int, const uint8_t *);
 static int callback_descend(void *);
+static int btree_node_fetch_canfail(struct btree * T, struct node * N,
+    int (* callback)(void *), void * cookie, int canfail);
```
Comment (cperciva) [inline kvlds/btree_node.c]: Don't need a forward declaration since this the function is only used after it's defined.

**Comment by gperciva** [comment]
Oops, fixed.  (the function still needs to be static, though)

**Comment by cperciva** [comment]
Looks good, please rebase.

**Comment by gperciva** [comment]
Updated with rebased history, ready for merge.

**Pull request merged**